packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/SuccessTokenLongShortPairFinancialProductLibrary.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./LongShortPairFinancialProductLibrary.sol";
+import "../../../../common/implementation/Lockable.sol";
+
+/**
+ * @title Success Token Long Short Pair Financial Product Library.
+ * @notice Adds settlement logic to create success token LSPs. A success token pays out a fixed amount of
+ * collateral as a floor, with the remaining amount functioning like an embedded option. The embedded
+ * option in this case uses payout logic that resembles a covered call. I.e., the token expires to be worth
+ * basePercentage + (1 - basePercentage) * (expiryPrice - strikePrice).
+ */
+contract SuccessTokenLongShortPairFinancialProductLibrary is LongShortPairFinancialProductLibrary, Lockable {
+    using FixedPoint for FixedPoint.Unsigned;
+
+    struct SuccessTokenLongShortPairParameters {
+        uint256 strikePrice;
+        uint256 basePercentage;
+    }
+
+    mapping(address => SuccessTokenLongShortPairParameters) public longShortPairParameters;
+
+    /**
+     * @notice Enables any address to set the strike price for an associated LSP.
+     * @param longShortPair address of the LSP.
+     * @param strikePrice the strike price for the covered call for the associated LSP.
+     * @param basePercentage the base percentage of collateral per pair paid out to long tokens, expressed
+     * with 1e18 decimals. E.g., a 50% base percentage should be expressed 500000000000000000, or 0.5 with
+     * 1e18 decimals. The base percentage cannot be set to 0.
+     * @dev Note: a) Any address can set the initial strike price b) A strike price cannot be 0.
+     * c) A strike price can only be set once to prevent the deployer from changing the strike after the fact.
+     * d) For safety, a strike price should be set before depositing any synthetic tokens in a liquidity pool.
+     * e) longShortPair must expose an expirationTimestamp method to validate it is correctly deployed.
+     */
+    function setLongShortPairParameters(
+        address longShortPair,
+        uint256 strikePrice,
+        uint256 basePercentage
+    ) public nonReentrant() {
+        require(ExpiringContractInterface(longShortPair).expirationTimestamp() != 0, "Invalid LSP address");
+        SuccessTokenLongShortPairParameters memory params = longShortPairParameters[longShortPair];
+        require(params.strikePrice == 0 && params.basePercentage == 0, "Parameters already set");
+        require(strikePrice != 0 && basePercentage != 0, "Base percentage and strike price cannot be set to 0");
```

Comment:
One other question: this require will let you set `strikePrice` or `basePercentage` to 0 (assuming the other is not zero). is there anything bad that can happen in this case?

If `strikePrice` is zero then the SuccessToken is always in the money. if `basePercentage` is zero this acts just like an option. Neither of these seem "bad" but just wanted to see if there is anything else to think about in this parameterization context.

Reply:
Hmm, I'm surprised that this would allow one parameter to be set to zero, since I thought the `&&` requires both to be non-zero?